macros/src/main/scala/me/rexim/morganey/meta/QuotationMacro.scala

```diff
@@ -134,23 +124,16 @@ private[meta] class QuotationMacro(val c: Context) {
         (q"_root_.scala.Some((..$ys))", q"_root_.scala.None")
     }
 
-    if (!hasSubpatterns() && ps.size == 1) {
-      q"""
-        new {
-          def unapply(input: $LambdaTermTpe) = Option(input)
-        }.unapply(..$args)
-      """
-    } else {
-      q"""
-        new {
-          def unapply(input: $LambdaTermTpe) = input match {
-            case $term => $ifp
-            case _     => $elp
-          }
-        }.unapply(..$args)
-      """
-    }
-
+    val identName = TermName(c.freshName())
+    q"""
+      new {
+        val $identName = true
```

Comment:
Some google results on the warning message http://alvinalexander.com/scala/scala-unreachable-code-due-to-variable-pattern-message